iree/hal/cuda/dynamic_symbols.c

```diff
@@ -53,6 +74,16 @@ iree_status_t iree_hal_cuda_dynamic_symbols_initialize(
         IREE_STATUS_UNAVAILABLE,
         "CUDA runtime library not available; ensure installed and on path");
   }
+  status = iree_dynamic_library_load_from_files(
```

Comment:
make this conditional - loading cupti is not free and can have side effects and we only want it enabled when we are going to use it. since it's only used for tracing you can guard this with `#if IREE_TRACING_FEATURES`